### DISCUSSION TITLE
Multi error messages

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#54898] [Q&A] [answered]
![error](https://github.com/symfony/symfony/assets/68124714/9d1bfec1-f7f2-4253-b1cf-5158a0fe81a9)
### Symfony version(s) affected

7.0.7

### Description

Hello. When I have a form and I have defined validation for this form, when I press the submit button repeatedly, the error is displayed as many times as the button is pressed!

### How to reproduce

twig:
```
{% for message in app.flashes('error') %}
    <div class="error-message">
        <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path d="M12 2C6.49 2 2 6.49 2 12C2 17.51 6.49 22 12 22C17.51 22 22 17.51 22 12C22 6.49 17.51 2 12 2ZM15.36 14.3C15.65 14.59 15.65 15.07 15.36 15.36C15.21 15.51 15.02 15.58 14.83 15.58C14.64 15.58 14.45 15.51 14.3 15.36L12 13.06L9.7 15.36C9.55 15.51 9.36 15.58 9.17 15.58C8.98 15.58 8.79 15.51 8.64 15.36C8.35 15.07 8.35 14.59 8.64 14.3L10.94 12L8.64 9.7C8.35 9.41 8.35 8.93 8.64 8.64C8.93 8.35 9.41 8.35 9.7 8.64L12 10.94L14.3 8.64C14.59 8.35 15.07 8.35 15.36 8.64C15.65 8.93 15.65 9.41 15.36 9.7L13.06 12L15.36 14.3Z" fill="#ef4444"/>
        </svg>
        <span class="ml-3">{{ message }}</span>
    </div>
{% endfor %}
```

controller:
```
...
$form = $formFactory->createNamed('', UserType::class, $user);
$form->handleRequest($request);
if ($form->isSubmitted() && $form->isValid()) {
    /** @var User $user */
    $user = $form->getData();
    $email = $user->getEmail();
    $checkEmail = $userRepository->findBy(['email' => $email]);
    if ($checkEmail) {
        $this->addFlash('error', 'Error message');

        return $this->redirectToRoute('app_check_email', [
            'id' => $id
        ]);
    }
...
```

**Comment by karaka200** [comment]
I checked now that in every form I have on the site, if I complete the fields and press the submit button 50 times, for example, a comment will be entered on the site 50 times or a post will be placed on the site 50 times, and when you check the database I see that 50 posts are placed in it. I followed all the things from CSRF, form submission type and validations. How can I fix this problem?

**Comment by MatTheCat** [comment, marked as answer]
Hello, you just discovered the infamous “double submit problem”. The usual work-around is to disable the form one way or another once it’s submitted; see https://www.bram.us/2020/11/04/preventing-double-form-submissions/ e.g.

**Reply by karaka200** [reply]
Thank you. Do you have a solution for me to create a limit on the server side?

**Reply by MatTheCat** [reply]
The issue with tackling this server-side is that you have to handle the first request (because you don’t know if more will follow), but the response reaching the user will be the last request’s.

That being said given how old this problem is I think solutions exist and people implemented them, but I’m not aware of any of them or their cost.
